api/src/pcapi/core/offers/offer_metadata.py

```diff
@@ -72,6 +80,50 @@ def _get_event_metadata_from_offer(offer: offers_models.Offer) -> Metadata:
     return common_metadata | event_metadata
 
 
+def _get_book_metadata_from_offer(offer: offers_models.Offer) -> Metadata:
+    common_metadata = _get_common_metadata_from_offer(offer)
+
+    book_metadata = {
+        "@type": "Book",
+        "@id": offer_app_link(offer),
+        "isFamilyFriendly": not offer.is_forbidden_to_underage,
+        "url": offer_app_link(offer),
+    }
+
+    work_example = {
+        "@type": "Book",
+        "@id": offer_app_link(offer),
+        "inLanguage": "fr",
+    }
+
+    extra_data = offer.extraData
+    if extra_data:
+        author = extra_data.get("author")
+        ean = extra_data.get("ean")
+        if author:
+            book_metadata["author"] = {
+                "@type": "Person",
+                "name": author,
+            }
+        if ean:
+            work_example["isbn"] = ean
+
+    if extra_data and extra_data.get("bookFormat") == offers_models.BookFormat.POCHE.value:
+        work_example["bookFormat"] = "https://schema.org/Paperback"
+    else:
+        match offer.subcategoryId:
+            case subcategories_v2.LIVRE_AUDIO_PHYSIQUE.id | subcategories_v2.TELECHARGEMENT_LIVRE_AUDIO.id:
+                work_example["bookFormat"] = "https://schema.org/AudiobookFormat"
+            case subcategories_v2.LIVRE_NUMERIQUE.id:
+                work_example["bookFormat"] = "https://schema.org/EBook"
+            case subcategories_v2.LIVRE_PAPIER.id:
+                work_example["bookFormat"] = "https://schema.org/Hardcover"
+
+    book_metadata["workExample"] = work_example
+
+    return common_metadata | book_metadata
```

Comment:
[détail / style] La définition de `common_metadata` est loin, au tout début de la fonction et n'est utilisée ici. Ce serait plus claire de définir cette variable juste avant son utilisation... ou même de ne pas définir de variable : 

```suggestion
    return _get_common_metadata_from_offer(offer) | book_metadata
```